### PR TITLE
Model stepped signals by constant interpolation of CombiTimeTable in MSL examples

### PR DESCRIPTION
**Pull request by beutlich**
Model stepped signals by constant interpolation of CombiTimeTable and avoid misuse of linear interpolation with duplicated sample points.

See #3115.

### CODE REVIEW

File: Modelica/Electrical/Machines.mo [hunk 1]
```diff
@@ -1168,9 +1168,9 @@ and accelerating inertias.<br>At time tStep a load step is applied.</p>
           annotation (Placement(transformation(
               extent={{-10,10},{10,-10}},
               rotation=270)));
-        Blocks.Sources.CombiTimeTable
-                                 dutyCycle(table=[0,0; 1,1; 4,1; 5,0; 10,0; 11,
-              -1; 14,-1; 15,0; 20,0], extrapolation=Modelica.Blocks.Types.Extrapolation.Periodic)
+        Blocks.Sources.CombiTimeTable dutyCycle(
```
Comment (HansOlsson) [inline Modelica/Electrical/Machines.mo]: To me the change for dutyCycle just looks like formatting change; am I missing something?

File: Modelica/Magnetic/FundamentalWave/Examples/BasicMachines/InductionMachines/IMC_Conveyor.mo [hunk 2]
```diff
@@ -41,8 +41,9 @@ model IMC_Conveyor
     annotation (Placement(transformation(extent={{60,10},{40,30}})));
   Modelica.Electrical.Machines.Sensors.CurrentQuasiRMSSensor currentQuasiRMSSensor
     annotation (Placement(transformation(extent={{20,100},{40,80}})));
-  Modelica.Blocks.Sources.CombiTimeTable dutyCycle(table=[0,0; 1,1; 4,1; 5,0;
-        10,0; 11,-1; 14,-1; 15,0; 20,0], extrapolation=Modelica.Blocks.Types.Extrapolation.Periodic)
+  Modelica.Blocks.Sources.CombiTimeTable dutyCycle(
```
Comment (HansOlsson) [inline Modelica/Magnetic/FundamentalWave/Examples/BasicMachines/InductionMachines/IMC_Conveyor.mo]: To me the change for dutyCycle just looks like formatting change.

File: Modelica/Magnetic/QuasiStatic/FundamentalWave/Examples/BasicMachines/InductionMachines/IMC_Conveyor.mo [hunk 3]
```diff
@@ -42,8 +42,9 @@ model IMC_Conveyor "Induction machine with squirrel cage and inverter driving a
     annotation (Placement(transformation(extent={{60,10},{40,30}})));
   Modelica.Electrical.QuasiStatic.Polyphase.Sensors.CurrentQuasiRMSSensor currentQuasiRMSSensorQS(m=m)
     annotation (Placement(transformation(extent={{20,100},{40,80}})));
-  Modelica.Blocks.Sources.CombiTimeTable dutyCycle(table=[0,0; 1,1; 4,1; 5,0;
-        10,0; 11,-1; 14,-1; 15,0; 20,0], extrapolation=Modelica.Blocks.Types.Extrapolation.Periodic)
+  Modelica.Blocks.Sources.CombiTimeTable dutyCycle(
```
Comment (HansOlsson) [inline Modelica/Magnetic/QuasiStatic/FundamentalWave/Examples/BasicMachines/InductionMachines/IMC_Conveyor.mo]: To me the change for dutyCycle just looks like formatting change.

File: Modelica/Mechanics/MultiBody/Examples/Elementary/RollingWheelSetPulling.mo [hunk 4]
```diff
@@ -28,11 +28,11 @@ model RollingWheelSetPulling "Rolling wheel set that is pulled by a force"
   Modelica.Mechanics.MultiBody.Parts.Body body(m=0.01, r_CM={0,0,0},
     animation=false)
     annotation (Placement(transformation(extent={{60,-10},{80,10}})));
-  Modelica.Blocks.Sources.CombiTimeTable combiTimeTable(table=[0,1,0,0; 1,1,
-        0,0; 2,0,2,0; 3,0,2,0])
+  Modelica.Blocks.Sources.CombiTimeTable combiTimeTable(
+    table=[0,1,0,0; 1,1,0,0; 2,0,2,0; 3,0,2,0])
```
Comment (HansOlsson) [inline Modelica/Mechanics/MultiBody/Examples/Elementary/RollingWheelSetPulling.mo]: To me the change for combiTimeTable just looks like formatting change (but an improvement).

File: Modelica/Mechanics/MultiBody/Examples/Elementary/RollingWheelSetPulling.mo [hunk 5]
```diff
@@ -28,11 +28,11 @@ model RollingWheelSetPulling "Rolling wheel set that is pulled by a force"
...
     annotation (Placement(transformation(extent={{-20,30},{0,50}})));
   Modelica.Mechanics.MultiBody.Parts.FixedTranslation fixedTranslation(
-                       r={0.2,0,0},
+    r={0.2,0,0},
```
Comment (HansOlsson) [inline Modelica/Mechanics/MultiBody/Examples/Elementary/RollingWheelSetPulling.mo]: To me the change for fixedTranslation just looks like formatting change (but an improvement).

File: Modelica/Electrical/Machines.mo (hunk 1 above)
Comment (beutlich) [reply Modelica/Electrical/Machines.mo]: Yes, cosmetics only. SimulationX has some issue when displaying arrays with line breaks:

grafik

**Comment by beutlich** [comment]
> as I don't see the need for all of the changes.

@HansOlsson OK for you to keep these changes in anyway?

**Comment by HansOlsson** [comment]
> as I don't see the need for all of the changes.
> 
> @HansOlsson OK for you to keep these changes in anyway?

Yes.

**Comment by beutlich** [comment]
@christiankral OK for you, too?

**Pull request merged**